system/jhtree/jhtree.cpp

```diff
@@ -1541,9 +1545,7 @@ const CJHSearchNode *CKeyIndex::locateLastNode(KeyStatsCollector &stats)
         depth++;
         NodeType type = (depth < getBranchDepth()) ? NodeBranch : NodeLeaf;
         cur = getNode(cur->nextNodeFpos(), type, stats.ctx);
-        //Unusual - an index with no elements
-        if (!cur)
-            return prev;
+        assertex(cur);
```

Comment:
This could only be null if the top branch node had 0 records.  I'm not convinced indexes should be generated with a single branch and no leaves, but I suspect some of the code in older versions of the platform relies on the top node being a branch, so I'm inclined to leave it as it is.